packages/database/src/migrations/20201117053359-AddPsssWoWIncreaseIndicators-modifies-data.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+import { arrayToDbString, generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const CONDITION_CODES = ['AFR', 'DIA', 'ILI', 'PF', 'DLI'];
+
+// eg 'PSSS_Confirmed_AFR_Site_Average'
+const psssCode = (descriptor, condition = '', isConfirmed = false) => {
+  const parts = ['PSSS', descriptor];
+  if (condition) {
+    parts.splice(1, 0, condition);
+  }
+  if (isConfirmed) {
+    parts.splice(1, 0, 'Confirmed');
+  }
+
+  return parts.join('_');
+};
+
+const insertIndicator = async (db, indicator) => {
+  await insertObject(db, 'indicator', { ...indicator, id: generateId() });
+  await insertObject(db, 'data_source', {
+    id: generateId(),
+    code: indicator.code,
+    type: 'dataElement',
+    service_type: 'indicator',
+  });
+};
+
+const createTotalSitesReportedIndicator = db =>
+  insertIndicator(db, {
+    code: psssCode('Total_Sites_Reported'),
+    builder: 'arithmetic',
+    config: {
+      // Just use a single data element for now, this logic will be extended in the future
+      formula: psssCode('Sites_Reported'),
+      aggregation: 'FINAL_EACH_WEEK',
+    },
+  });
+
+const createTotalCasesIndicator = async (db, condition) =>
+  insertIndicator(db, {
+    code: psssCode('Total_Cases', condition),
+    builder: 'arithmetic',
+    config: {
+      // Just use a single data element for now, this logic will be extended in the future
+      formula: psssCode('Cases', condition),
+      aggregation: 'FINAL_EACH_WEEK',
+    },
+  });
+
+const createSiteAverageIndicator = async (db, condition, isConfirmed) => {
+  const cases = psssCode(isConfirmed ? 'Cases' : 'Total_Cases', condition, isConfirmed);
+  const sitesReported = psssCode(isConfirmed ? 'Sites_Reported' : 'Total_Sites_Reported');
+
+  return insertIndicator(db, {
+    code: psssCode('Site_Average', condition, isConfirmed),
+    builder: 'arithmetic',
+    config: {
+      formula: `${cases} / ${sitesReported}`,
+      aggregation: 'FINAL_EACH_WEEK',
+    },
+  });
+};
+
+const createWowIncreaseIndicator = async (db, condition, isConfirmed) => {
+  const currentAverage = psssCode('Site_Average', condition, isConfirmed);
+  const prevAverage = 'siteAveragePrevWeek';
```

Comment:
It may be useful to separate "data source" and "parameter" variables in a formula by using a different conventions. Since we use SNAKE_CASE for data sources, we could use cameCase for parameters. The formula will look like:

`PSSS_AFR_Site_Average / siteAveragePrevWeek`